# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 2424d5c30267
# Full Hash: 2424d5c30267f67ab2b28119d75d89f0441b30e0
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2024-01-23 05:36:48
# Regressor Bug: 1833814
# File Overlap Count: 1
# Description:
#   Bug 1833814 - change nsIWidget's cursor logic, r=emilio,win-reviewers,mhowell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195549
# ==============================================================================

diff -r 94f46aad00df -r 2424d5c30267 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Mon Jan 22 20:10:54 2024 +0000
+++ b/widget/gtk/nsWindow.cpp	Mon Jan 22 20:42:50 2024 +0000
@@ -3328,7 +3328,10 @@
   mCursor = aCursor;
 
   // Try to set the cursor image first, and fall back to the numeric cursor.
-  GdkCursor* imageCursor = GetCursorForImage(aCursor, GdkCeiledScaleFactor());
+  GdkCursor* imageCursor = nullptr;
+  if (mCustomCursorAllowed) {
+    imageCursor = GetCursorForImage(aCursor, GdkCeiledScaleFactor());
+  }
 
   // When using a custom cursor, clear the cursor first using eCursor_none, in
   // order to work around https://gitlab.gnome.org/GNOME/gtk/-/issues/6242